plugins/openshift/commands/crd-review.md

```diff
@@ -0,0 +1,277 @@
+
+### Kubernetes API Conventions
+
+#### Naming Conventions
+- **Resource Names**: Must follow DNS label format (lowercase, alphanumeric, hyphens)
+- **Field Names**: PascalCase for Go, camelCase for JSON
+- **Avoid**: Abbreviations, underscores, ambiguous names
+- **Include**: Units/types in field names when needed (e.g., `timeoutSeconds`)
+
+#### API Structure
+- **Required Fields**: Every object must have `kind` and `apiVersion`
+- **Metadata**: Should include `namespace`, `name`, `uid`, optionally `resourceVersion`, `labels`, `annotations`
+- **Spec/Status Separation**: Clear separation between desired state (spec) and observed state (status)
+
+#### Status Field Design
+- **Conditions**: Must include conditions array with:
+  - `type`: Clear, human-readable condition type
+  - `status`: `True`, `False`, or `Unknown`
+  - `reason`: Machine-readable reason code
+  - `message`: Human-readable message
+  - `lastTransitionTime`: RFC 3339 timestamp
+
+#### Field Types
+- **Integers**: Prefer `int32` over `int64`
+- **Avoid**: Unsigned integers, floating-point values
+- **Enums**: Use string constants, not numeric values
+- **Optional Fields**: Use pointers in Go
+
+#### Versioning
+- **Group Names**: Use domain format (e.g., `myapp.example.com`)
+- **Version Strings**: Must match DNS label format (e.g., `v1`, `v1beta1`)
+- **Migration**: Provide clear paths between versions
+
+### OpenShift API Conventions
+
+#### Configuration vs Workload APIs
+- **Configuration APIs**: Typically cluster-scoped, manage cluster behavior
+- **Workload APIs**: Usually namespaced, user-facing resources
+
+#### Field Design
+- **Avoid Boolean Fields**: Use enumerations instead of binary true/false
+  - ❌ Bad: `enabled: true`
+  - ✅ Good: `state: "Enabled"` with enum values `["Enabled", "Disabled", "Auto"]`
+- **Object References**: Use specific types, omit "Ref" suffix
+- **Clear Semantics**: Each field should have one clear purpose
+
+#### Documentation Requirements
+- **Godoc Comments**: Comprehensive documentation for all exported types and fields
+- **JSON Field Names**: Use JSON names in documentation (not Go names)
+- **User-Facing**: Write for users, not just developers
+- **Explain Interactions**: Document how fields interact with each other
+
+#### Validation
+- **Kubebuilder Tags**: Use validation markers (`+kubebuilder:validation:*`)
+- **Enum Values**: Explicitly define allowed values
+- **Field Constraints**: Define minimums, maximums, patterns
+- **Meaningful Errors**: Validation messages should guide users
+
+#### Union Types
+- **Discriminated Unions**: Use a discriminator field to select variant
+- **Optional Pointers**: All union members should be optional pointers
+- **Validation**: Ensure exactly one union member is set
+
+## Implementation
+
+The command performs the following analysis workflow:
+
+1. **Repository Discovery**
+   - Scan repository for CRD files (YAML manifests in `config/`, `deploy/`, `manifests/` directories)
```

Comment:
This Implementation section promises concrete analysis (discover CRDs, validate conventions, assign severities), but it never gives the agent actionable steps or heuristics to execute. Without instructions on how to locate CRD YAML, walk spec.versions[].schema, inspect Go types, or derive severities, the command will just restate guidelines instead of performing a real review. I think you should flesh this section out with the actual workflow (e.g., how to enumerate CRD files, what to look for when checking naming/spec/status, how to collect file & line references) so the agent can deliver the promised analysis